### PR TITLE
[Refactor] Code readability update

### PR DESCRIPTION
**Pull request by DayKev**
## What are the changes?
Some non-functional changes are made to improve the readability of the code in `src/field/pokemon.ts` (this code provided by DerTapp on Discord) and `src/data/pokemon-species.ts` (by me).

## Why am I doing these changes?
A convoluted nested ternary was noticed while looking through the code.

## What did change?
Nested ternary replaced with commented if/else blocks in `src/field/pokemon.ts`.
Nested ternaries in `PokemonSpeciesForm.getAbilityCount()` and `PokemonSpeciesForm.getAbility()` changed to if/then blocks in `src/data/pokemon-species.ts`. These weren't that bad, but I'm not a fan of nested ternaries

## How to test the changes?
Uh... play the game? All of the tests still pass and this should theoretically be entirely non-functional changes.

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- ~[ ] Are the changes visual?~
  - ~[ ] Have I provided screenshots/videos of the changes?~

### CODE REVIEW

**Review by CodeTappert — APPROVED** [review]

File: src/data/pokemon-species.ts [hunk 1]
```diff
@@ -178,11 +178,30 @@ export abstract class PokemonSpeciesForm {
   }
 
   getAbilityCount(): integer {
-    return this.ability2 ? this.abilityHidden ? 3 : 2 : this.abilityHidden ? 2 : 1;
+    let count = 1;
+    if (this.ability2) {
```
Comment (mcmontag) [inline src/data/pokemon-species.ts:182]: In `getAbility`, `ability2` is checked to make sure it's not `Abilities.NONE`.  I'm assuming `Abilities.NONE` has an actual value of `0`, so the two conditions are functionally equivalent, but it feels like they should be written the same way.

**Comment by flx-sta** [comment]
If you are refactoring code, please write automated tests for the refactored section to make sure that you didn’t change any expected results

File: src/data/pokemon-species.ts (hunk 1 above)
Comment (DayKev) [reply src/data/pokemon-species.ts:182]: Whoops, missed that one, ty!

File: src/data/pokemon-species.ts [hunk 2]
```diff
@@ -178,11 +178,30 @@ export abstract class PokemonSpeciesForm {
   }
 
   getAbilityCount(): integer {
```
Comment (flx-sta) [inline src/data/pokemon-species.ts:180]: Please add some jsdoc

File: src/data/pokemon-species.ts [hunk 3]
```diff
@@ -178,11 +178,30 @@ export abstract class PokemonSpeciesForm {
...
+    if (this.ability2 !== Abilities.NONE) {
+      count += 1;
+    }
+    if (this.abilityHidden !== Abilities.NONE) {
+      count += 1;
+    }
+    return count;
   }
 
   getAbility(abilityIndex: integer): Abilities {
```
Comment (flx-sta) [inline src/data/pokemon-species.ts:191]: Please add some jsdoc

File: src/test/internals.test.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+  afterEach(() => {
+    game.phaseInterceptor.restoreOg();
+  });
+
+  beforeEach(() => {
+    game = new GameManager(phaserGame);
+  });
+
+  it("PokemonSpeciesForm - .getAbilityCount() and .getAbility()", async () => {
```
Comment (flx-sta) [inline src/test/internals.test.ts:25]: Test names should be a "narrative". The "it" is part of it.

e.g. "it should provide Eevee with 3 defined abilities"
=> 
```suggestion
  it("it should provide Eevee with 3 defined abilities", async () => {
```

File: src/test/internals.test.ts [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    const eevee = game.scene.getPlayerPokemon();
+
+    expect(eevee.getSpeciesForm().getAbilityCount()).toBe(3);
+
+    expect(eevee.getSpeciesForm().getAbility(0)).toBe(Abilities.RUN_AWAY);
+    expect(eevee.getSpeciesForm().getAbility(1)).toBe(Abilities.ADAPTABILITY);
+    expect(eevee.getSpeciesForm().getAbility(2)).toBe(Abilities.ANTICIPATION);
+  });
+
+  it("Pokemon.abilityIndex definition", async () => {
```
Comment (flx-sta) [inline src/test/internals.test.ts:36]: Test names should be a "narrative". The "it" is part of it.

e.g. "it should set Eeeve abilityIndex between 0-2"
=> 
```ts
```suggestion
  it("should set Eeeve abilityIndex between 0-2", async () => {
```

File: src/test/internals.test.ts [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    await game.runToSummon([Species.EEVEE]);
+    const eevee = game.scene.getPlayerPokemon();
+
+    const abilityIndexInRange = ([0, 1, 2].includes(eevee.abilityIndex));
+    expect(abilityIndexInRange).toBe(true);
```
Comment (flx-sta) [inline src/test/internals.test.ts:41]: expect(eevee.abilityIndex).toBeGreaterThanOrEqual(0);
    expect(eevee.abilityIndex).toBeLessThanOrEqual(2);

**Review by flx-sta — CHANGES_REQUESTED** [review]

**Review by flx-sta — APPROVED** [review]

**Review by CodeTappert — APPROVED** [review]

**Pull request merged**